Fix label `for` attribute when not scoped to model

myabc opened the pull request:
### What are you trying to accomplish?

Defers assigning label `for` value to later in `initialize` method, ensuring the `scope_id_false: false` option is respected.

### Integration

It may require updating calling code that relies on the incorrect behavior.

#### List the issues that this change affects.

Closes primer/view_components#3695

#### Risk Assessment

- [ ] **Low risk** the change is small, highly observable, and easily rolled back.
- [X] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.


### Merge checklist

- [X] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

myabc committed:
Fix label `for` attribute when not scoped to model

Defers assigning label `for` value to later in `initialize` method,
ensuring the `scope_id_false: false` option is respected.

Closes primer/view_components#3695

myabc committed:
Add changeset

jonrohan commented on app/lib/primer/forms/dsl/input.rb:
```diff
@@ -58,7 +58,6 @@ def initialize(builder:, form:, **system_arguments)
           @input_arguments = system_arguments
           @input_arguments.delete(:id) unless @input_arguments[:id].present?
           @label_arguments = @input_arguments.delete(:label_arguments) || {}
-          @label_arguments[:for] = id if id.present?
```
This was calling `def id` below which is the same as `@input_arguments[:id]` curious how this is different than what was already in place

myabc committed:
Guard label `for` against nil input id

Setting `for: nil` made Rails skip generating the scoped id, dropping
the label association on the default model-scoped path. Sets `for` only
when the input id is present, restoring Rails autogeneration otherwise.
Adds a default-scope regression test and corrects the changeset to name
the real `scope_id_to_model` option.

myabc commented:
@jonrohan took a moment to revisit this. Unfortunately my solution was missing a `if @input_arguments[:id].present?` guard (which could have caused another regression). Fixed in [4a06e61](https://github.com/primer/view_components/pull/3696/commits/4a06e618e19213856e421bc3ffbd73c337656624)
